src/main/kotlin/org/jitsi/jicofo/xmpp/XmppServices.kt

```diff
@@ -45,6 +58,11 @@ class XmppServices(xmppProviderFactory: XmppProviderFactory) {
         setOf(clientConnection.xmppConnection, serviceConnection.xmppConnection)
     )
 
+    val avModerationHandler = AvModerationHandler(clientConnection.xmppConnection).also {
+        clientConnection.xmppConnection.addAsyncStanzaListener(it, MessageTypeFilter.NORMAL)
+        clientConnection.addRegistrationListener(it)
```

Comment:
@bgrozev I was wondering is it possible for the connection to be already registered when we do this and skip the disco-info?

Reply:
I added a check for that ... not sure how good it is with no synchronization ...

Reply:
Aaaand I broke the tests with that :)

Reply:
```
SEVERE: Error checking for av_moderation component
org.jivesoftware.smack.SmackException$NoResponseException: No response received within reply timeout. Timeout was 15000ms (~15s). Waited for response using: IQReplyFilter: iqAndIdFilter (AndFilter: (OrFilter: (IQTypeFilter: type=error, IQTypeFilter: type=result), StanzaIdFilter: id=hP27l-141)), : fromFilter (OrFilter: (FromMatchesFilter (full): test.domain.net)).
	at org.jivesoftware.smack.SmackException$NoResponseException.newWith(SmackException.java:111)
	at org.jivesoftware.smack.SmackException$NoResponseException.newWith(SmackException.java:98)
	at org.jivesoftware.smack.StanzaCollector.nextResultOrThrow(StanzaCollector.java:260)
	at org.jivesoftware.smack.StanzaCollector.nextResultOrThrow(StanzaCollector.java:214)
	at org.jivesoftware.smackx.disco.ServiceDiscoveryManager.discoverInfo(ServiceDiscoveryManager.java:540)
	at org.jivesoftware.smackx.disco.ServiceDiscoveryManager.discoverInfo(ServiceDiscoveryManager.java:506)
	at org.jitsi.jicofo.xmpp.AvModerationHandler.registrationChanged(XmppServices.kt:154)
	at org.jitsi.jicofo.xmpp.XmppServices.<init>(XmppServices.kt:64)
```

@bgrozev as you are playing with the tests, is there a way to skip this step when testing?

Reply:
I dropped that commit. 

But now I see Mute test is broken, so there must be some problem ... will be checking that.

Reply:
And that's only in FF case ... hum is this what you were seeing yesterday for some other PR?

Reply:
Oh that is a problem in the bridge not forwarding the video stream, and as audio is muted nothing is received and it fails

Reply:
Yep, retrying it, worked, it is now just the known desktop sharing one.

Reply:
> @bgrozev I was wondering is it possible for the connection to be already registered when we do this and skip the disco-info?

Yes, it is. For now you can add something like `it.registrationChanged(clientConnection.isRegistered())` which should make the probability of the race very low. I'll probably change this soon and I'll have it in mind.

Reply:
I added it, but it breaks the test. Will leave it there commented.